NebulaModel/DataStructures/PlayerData.cs

```diff
@@ -63,6 +80,20 @@ public void Deserialize(INetDataReader reader)
             BodyRotation = reader.GetFloat3();
             Mecha = new MechaData();
             Mecha.Deserialize(reader);
+            bool isAppearancePresent = reader.GetBool();
+            if (isAppearancePresent)
+            {
+                int len = reader.GetInt();
+                byte[] data = new byte[len];
+                reader.GetBytes(data, len);
+                using (MemoryStream ms = new MemoryStream(data))
+                using (BinaryReader br = new BinaryReader(ms))
+                {
+                    Appearance = new MechaAppearance();
+                    Appearance.Init();
+                    Appearance.Import(br);
+                }
```

Comment:
Can't you use `Appearance.FromBytes(data)`?

Reply:
Again, yes. But its just another way to write the same thing.